skimage/segmentation/active_contour_model.py

```diff
@@ -0,0 +1,204 @@
+        edges.
+    gamma : float, optional
+        Excpliti time stepping parameter.
+    bc : {'periodic', 'free', 'fixed'}, optional
+        Boundary conditions for worm. 'periodic' attaches the two ends of the
+         snake, 'fixed' holds the end-points in place, and'free' allows free
+         movement of the ends. 'fixed' and 'free' can be combined by parsing
+         'fixed-free', 'free-fixed'. Parsing 'fixed-fixed' or 'free-free'
+         yields same behaviour as 'fixed' and 'free', respectively.
+    max_px_move : float, optional
+        Maximum pixel distance to move per iteration.
+    max_iterations : int, optional
+        Maximum iterations to optimize snake shape.
+    convergence: float, optional
+        Convergence criteria.
+
+    Returns
+    -------
+    snake : (N, 2) ndarray
+        Optimised snake, same shape as input parameter.
+
+    References
+    ----------
+    .. [1]  Kass, M.; Witkin, A.; Terzopoulos, D. "Snakes: Active contour
+    models". International Journal of Computer Vision 1 (4): 321 (1988).
+
+    Examples
+    --------
+    >>> from skimage.draw import circle_perimeter
+    >>> from skimage.filters import gaussian_filter
+    Create and smooth image:
+    >>> img = np.zeros((100, 100))
+    >>> rr, cc = circle_perimeter(35, 45, 25)
+    >>> img[rr, cc] = 1
+    >>> img = gaussian_filter(img, 2)
+    Initiliaze spline:
+    >>> s = np.linspace(0, 2*np.pi,100)
+    >>> init = 50*np.array([np.cos(s), np.sin(s)]).T+50
+    Fit spline to image:
+    >>> snake = active_contour_model(img, init, w_edge=0, w_line=1)
+    >>> int(np.mean(np.sqrt((45-snake[:, 0])**2 + (35-snake[:, 1])**2)))
+    25
+
+    """
+
+    max_iterations = int(max_iterations)
+    if max_iterations <= 0:
+        raise ValueError("max_iterations should be >0.")
+    convergence_order = 10
+    valid_bcs = ['periodic', 'free', 'fixed', 'free-fixed',
+                 'fixed-free', 'fixed-fixed', 'free-free']
+    if bc not in valid_bcs:
+        raise ValueError("Invalid boundary condition.\n"+
+                         "Should be one of: "+", ".join(valid_bcs)+'.')
+    img = img_as_float(image)
+    RGB = img.ndim == 3
+
+    # Find edges using sobel:
+    if w_edge != 0:
+        if RGB:
+            edge = [sobel(img[:, :, 0]), sobel(img[:, :, 1]),
+                    sobel(img[:, :, 2])]
+        else:
+            edge = [sobel(img)]
+        for i in xrange(3 if RGB else 1):
+            edge[i][0, :] = edge[i][1, :]
+            edge[i][-1, :] = edge[i][-2, :]
+            edge[i][:, 0] = edge[i][:, 1]
+            edge[i][:, -1] = edge[i][:, -2]
+    else:
+        edge = [0]
+
+    # Superimpose intensity and edge images:
+    if RGB:
+        img = w_line*np.sum(img, axis=2) \
+            + w_edge*sum(edge)
+    else:
+        img = w_line*img + w_edge*edge[0]
+
+    # Interpolate for smoothness:
+    intp = RectBivariateSpline(np.arange(img.shape[1]),
```

Comment:
any interest in fixing the ``bbox`` Parameter of  ``RectBivariateSpline`` to model only a region of interest? (using a rectangle enclosing the initial worm). I haven't tried it, so I'm not sure it's worth the pain, but since most of the time is spent in calls to ``intp`` (as I saw from using the line profile of the astronaut example), maybe reducing the memory used by the object ``intp`` might speed up things a little bit for large images... I'm not sure, though.